### PR TITLE
Feat 138 add detailed document information to status page

### LINKED ISSUE #138: Display detailed information on Status page for each table row
**Is your feature request related to a problem? Please describe.**
As an authenticated user of SOLO, I want to be able to view detailed information about a given row on the status page by tapping an accordion button. 

**Describe the solution you'd like**

- Setup accordion functionality in table
- Build out extra data in expanded are including timeline component

### PR DESCRIPTION
**Pull request by jhackshaw-dds**
Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist
I have…
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] attached screenshots illustrating relevant behavior before and after my changes.
~~[ ] review and update SSAG documentation if needed.~~

## Summary of Changes
This pull request…
- closes #138 
- add details component that is rendered in table row expansion
- add details data from solo document
- add and update unit tests
- add ApiDocument type and an explicit conversion step after api call
- move page components into their own directory
- update solo-types with new expected response structure

## Testing
To verify the changes proposed in this pull request…
1. Run frontend unit tests
2. Run development environment and view status page

## Screenshots
image

### CODE REVIEW

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  part: Part[];
+}
+
+const StatusPageDetailData: React.FC<StatusPageDetailDataProps> = ({
+  address,
+  status,
+  part
+}) => {
+  const shipperAddy = address.find(addy => addy.address_type === "2")?.name;
+  const receiverAddy = address.find(addy => addy.address_type === "3")?.name;
```
Comment (SCasarottoDDS) [inline frontend/src/components/StatusPage/StatusPageDetailData.tsx]: I don't love the need to do a `.find` on every render. My first recommendation would be to only pass in the address needed and make the component dumber.  The hard coded address_type feels weird. If that isn't an option (I'd like to know why) at least memoize it (`useMemo`)

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  const quantity = status[0]?.qty || 0;
```
Comment (SCasarottoDDS) [inline frontend/src/components/StatusPage/StatusPageDetailData.tsx]: Just want to check my own assumptions. Is there a real benefit or difference between `|| 0` and `?? 0`

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx (hunk 1 above)
Comment (SCasarottoDDS) [reply frontend/src/components/StatusPage/StatusPageDetailData.tsx]: Ultimately what I would do is just hold this data as an object and look up by key.

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { formatDistanceToNow, parseISO } from "date-fns";
+import { Address, Status, Part } from "solo-types";
+
+interface StatusPageDetailDataProps {
+  address: Address[];
+  status: Status[];
+  part: Part[];
```
Comment (SCasarottoDDS) [inline frontend/src/components/StatusPage/StatusPageDetailData.tsx]: Prop naming feels weird. Feels like a singular thing when there arrays of stuff.

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          <div className="flex-col">
+            <div className="text-bold">Quantity</div>
+            <div>{quantity}</div>
+          </div>
+          <div className="flex-col">
+            <div className="text-bold">Unit of Measure</div>
+            <div>{part[0].uom}</div>
+          </div>
+        </>
+      )}
```
Comment (SCasarottoDDS) [inline frontend/src/components/StatusPage/StatusPageDetailData.tsx]: Feels weird to unwrap an array.  How does the user of this component know that the first spot in the array needs to be this info? Or does every item in this array have the same info?

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx (hunk 3 above)
Comment (SCasarottoDDS) [reply frontend/src/components/StatusPage/StatusPageDetailData.tsx]: After looking at this component I only think `status` should be an array and I think the same should change to something like `statuses` or `statusArray`

File: frontend/src/components/StatusPage/StatusPageDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  const { status, part, address } = document;
+
+  return (
+    <>
+      <StatusPageDetailData address={address} part={part} status={status} />
+    </>
+  );
+};
+
+export default StatusPageDetails;
```
Comment (SCasarottoDDS) [inline frontend/src/components/StatusPage/StatusPageDetails.tsx]: Why make this a separate component? OR why separate out the data component?

**Review by SCasarottoDDS — CHANGES_REQUESTED** [review]

File: frontend/src/components/StatusPage/StatusPageDetails.tsx (hunk 5 above)
Comment (jhackshaw-dds) [reply frontend/src/components/StatusPage/StatusPageDetails.tsx]: This is where the timeline component will go

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx (hunk 1 above)
Comment (jhackshaw-dds) [reply frontend/src/components/StatusPage/StatusPageDetailData.tsx]: Added explicit conversion step from the structure of the api response to a more frontend focused data structure.

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx (hunk 2 above)
Comment (jhackshaw-dds) [reply frontend/src/components/StatusPage/StatusPageDetailData.tsx]: Added explicit conversion step from the structure of the api response to a more frontend focused data structure.

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx (hunk 3 above)
Comment (jhackshaw-dds) [reply frontend/src/components/StatusPage/StatusPageDetailData.tsx]: @lit-dds changed the key names of the data in the api response, and an explicit conversion step from the structure of the api response to a more frontend focused data structure was added.

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx (hunk 4 above)
Comment (jhackshaw-dds) [reply frontend/src/components/StatusPage/StatusPageDetailData.tsx]: api is expected to return a single part object now.

File: frontend/src/components/StatusPage/StatusPageDetailData.tsx (hunk 2 above)
Comment (jhackshaw-dds) [reply frontend/src/components/StatusPage/StatusPageDetailData.tsx]: No longer necessary either way

File: frontend/src/components/DocumentDetails/DocumentDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+          <div>{shipper.name}</div>
+        </div>
+      )}
+      {receiver && (
+        <div className="flex-col">
+          <div className="text-bold">Shipped To</div>
+          <div>{receiver.name}</div>
+        </div>
+      )}
+    </div>
```
Comment (SCasarottoDDS) [inline frontend/src/components/DocumentDetails/DocumentDetails.tsx]: This is a UI/UX question. Should we display the spots with empty data to imply there isn't data for the user or should we hide them. I haven't thought much about this but would love your feedback.

Comment (SCasarottoDDS) [reply frontend/src/components/DocumentDetails/DocumentDetails.tsx]: What I mean by that is should the title be there no matter what.

File: frontend/src/pages/StatusPage/useDocuments.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import { useCallback, useEffect, useState } from "react";
+import { SortingRule } from "react-table";
+import useAuthContext from "context/AuthContext";
+import { Document, ApiDocument } from "solo-types";
+import { createFakeApiDocs } from "solo-types";
+
+// covert api returned document to Document
+export const parseApiDocuments = (apiDocs: ApiDocument[]): Document[] =>
```
Comment (SCasarottoDDS) [inline frontend/src/pages/StatusPage/useDocuments.ts:8]: Thoughts? Should there be tests for this and the `useDocuments`? 🤷‍♂

**Review by SCasarottoDDS — CHANGES_REQUESTED** [review]

File: frontend/src/pages/StatusPage/useDocuments.ts (hunk 7 above)
Comment (jhackshaw-dds) [reply frontend/src/pages/StatusPage/useDocuments.ts:8]: The behavior is tested tested completely by testing the StatusPage component:

image

I can add explicit tests too though

Comment (SCasarottoDDS) [reply frontend/src/pages/StatusPage/useDocuments.ts:8]: Nah thats fine. I expected that it was covered. Nbd its all pretty normal stuff.

**Review by SCasarottoDDS — APPROVED** [review]

File: frontend/src/components/DocumentDetails/DocumentDetails.tsx (hunk 6 above)
Comment (jhackshaw-dds) [reply frontend/src/components/DocumentDetails/DocumentDetails.tsx]: Agreed, updated.

**Pull request merged**